core/src/autogluon/core/utils/feature_selection.py

```diff
@@ -0,0 +1,497 @@
+        if self.is_bagged:
+            evaluated_df['n'] = evaluated_df['n'] // len(model.models)
+
+        # if we could not compute feature importance for all features and previous feature importance estimates exist, use them
+        importance_df = pd.concat([evaluated_df, unevaluated_fi_df_template(unevaluated_features)])
+        importance_df = merge_importance_dfs(prev_importance_df, importance_df, using_prev_fit_fi)
+
+        # if auto_threshold, threshold is the mean of noise column importance score
+        if auto_threshold:
+            noise_rows = importance_df[importance_df.index.isin(prioritized)]
+            importance_df = importance_df.drop(prioritized)
+            prune_threshold = noise_rows['importance'].mean()
+
+        # use importance_df to generate next candidate features
+        time_budget_select = time_budget - (time.time() - time_start)
+        candidate_features = self.compute_next_candidate_given_fi(importance_df, prune_threshold, prune_ratio, time_budget_select, weighted)
+
+        # if noise columns exist, they should never be removed
+        if auto_threshold:
+            candidate_features = candidate_features + list(prioritized)
+            importance_df = pd.concat([importance_df, noise_rows])
+
+        feature_selection_time = time.time() - time_start
+        self.time_limit = self.time_limit - feature_selection_time
+        self._fi_time_elapsed = self._fi_time_elapsed + feature_selection_time
+        return candidate_features, importance_df.sort_values(by='importance', axis=0), feature_selection_time
+
+    def compute_next_candidate_given_fi(self, importance_df: pd.DataFrame, prune_threshold: float, prune_ratio: float, time_budget: float,
+                                        weighted: bool) -> List[str]:
+        """
+        Keep features whose importance scores are above threshold or have not yet had a chance to be calculated,
+        as well as some features whose importance scores are below threshold if more than prune_ratio * num features
+        features are below threshold. In the latter case, randomly sample without replacement from features whose
+        importance scores are below threshold until removal candidate configuration that has not yet been tried
+        is encountered. Give higher probability to features whose scores are lower than others when sampling.
+        """
+        time_start = time.time()
+        n_remove = max(1, int(prune_ratio * len(importance_df)))
+        above_threshold_rows = importance_df[(importance_df['importance'] > prune_threshold) | (importance_df['importance'].isna())]
+        below_threshold_rows = importance_df[importance_df['importance'] <= prune_threshold].sort_values(by='importance', axis=0, ascending=True)
+        logger.log(30, f"\tNumber of identified features below prune threshold {round(prune_threshold, 4)}: {len(below_threshold_rows)}/{len(importance_df)}")
```

Comment:
statement related to internal operations should have lower logging level (probably 15)